### DISCUSSION TITLE
Does AIR SDK IOS development not support Video?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#2889] [Q&A]
I cannot play videos using VIDEO on IOS, is it not supported? Android can do it!

**Comment by itlancer** [comment]
@Jason0247 
MP4 H.264 video playback not supported with simple `Video` object for iOS: https://github.com/airsdk/Adobe-Runtime-Support/issues/211
Only FLV H.263 supported for `Video` with iOS.

`VideoTexture` have multiple issues like https://github.com/airsdk/Adobe-Runtime-Support/issues/1939 but may be it works for you.
`StageVideo` can be used for simple cases but it also have some issues https://github.com/airsdk/Adobe-Runtime-Support/issues/1942

**Reply by Jason0247** [reply]
@itlancer Thank you!

**Comment by Jason0247** [comment]
I want to know why IOS does not allow VIDEO objects to play H.264 videos? I think this feature is really ordinary and often requires some interaction. Is it not open for any reason?
thank

**Reply by ajwfrost** [reply]
Hi

I believe it should be possible for us to get this working, we're currently looking into the VideoTexture issues that @itlancer had raised earlier. The normal video objects have an additional complication whereby we need to be able to get at the decoded frame data, which is dependent upon the OS capabilities. In the past, Adobe had been using a software H.264 decoder which involved some special patent and software licensing that we don't have ourselves, hence having to find different ways to handle this.

Will talk with the team and see where we're at with these ones..

thanks

**Reply by Jason0247** [reply]
This is really good news. There are also some issues with Android, which is that only one H.264 video can be played, and the second one will not be displayed. I hope we can solve this together. Thank you again.

**Reply by D5CN** [reply]
i just found this,it's good news,MP4 will be back^^

**Comment by Jason0247** [comment]
Just saw someone posting that using stageVideo on iOS cannot play videos of MP4 H.264 because the license for MP4 has expired. Is that true? @ajwfrost 
#2933

**Reply by ajwfrost** [reply]
Sort of .. mp4 is the container, it's the codecs that are the problem. The AIR runtimes up to 32.0.89 contained software decoders for AAC and AVC (H.264) that Adobe had licensed and was paying royalties for. When they closed down the Flash Player deployments they wanted to stop those commitments hence the builds that we have don't include the software AAC/AVC decoding. This had been used across all platforms; Adobe had put in platform-specific alternatives - although these weren't without issues, so we've had to do a bit more to get the video decoding again on each platform. iOS is still a bit of a problem area for us that we're working on currently...

StageVideo though should always have been using the platform-specific (hardware accelerated, hopefully) decoders, so the above licensing/codec issues shouldn't have impacted that!
